scalding-core/src/main/scala/com/twitter/scalding/typed/Grouped.scala

```diff
@@ -365,8 +365,14 @@ object Grouped extends Serializable {
 
   def addEmptyGuard[K, V1, V2](fn: (K, Iterator[V1]) => Iterator[V2]): (K, Iterator[V1]) => Iterator[V2] =
     fn match {
-      case alreadyGuarded@EmptyGuard(_) => alreadyGuarded
-      case ami if CoGroupable.atMostInputSizeFn(ami) => ami // already safe
```

Comment:
yeah, I think the bug is that atMostInputSizeFn is just wrong for folds, which as you point out always emit 1 thing, which can be greater than 1.

Reply:
ok. let's go with this.